extension/EdFi.Ods.Extensions.LMSX/LMS-Harmonizer/1090-lms-assignment-submissions.sql

```diff
@@ -63,7 +123,7 @@ BEGIN
 		[Grade]
 	FROM #ALL_SUBMISSIONS
 	WHERE
-		#ALL_SUBMISSIONS.AssignmentSubmissionIdentifier NOT IN
+		#ALL_SUBMISSIONS.SourceSystemIdentifier NOT IN
 			(SELECT DISTINCT AssignmentSubmission.AssignmentSubmissionIdentifier FROM LMSX.AssignmentSubmission)
 		AND #ALL_SUBMISSIONS.StudentUSI NOT IN
             (SELECT DISTINCT StudentUSI FROM LMSX.AssignmentSubmission)
```

Comment:
I think this `DISTINCT` and the one on line 127 are not necessary and could cause some performance problems. We might want to rewrite with a "NOT EXISTS" type query instead.

Reply:
Actually lines 128 and 129 are just wrong. This would prevent a student from having two submission records! But they will have multiple assignments, so they need to have multiple submissions.

Reply:
```
    SELECT
        SourceSystemIdentifier,
        StudentUSI,
        AssignmentIdentifier,
        @Namespace,
        DescriptorId,
        SubmissionDateTime,
        EarnedPoints,
        Grade
	FROM #ALL_SUBMISSIONS
	WHERE
    NOT EXISTS (
        SELECT 1 FROM LMSX.AssignmentSubmission WHERE AssignmentSubmissionIdentifier = #ALL_SUBMISSIONS.SourceSystemIdentifier
    )
	AND
        #ALL_SUBMISSIONS.DeletedAt IS NULL
```